src/common/src/types/mod.rs

```diff
@@ -321,6 +325,31 @@ impl DataType {
             .into(),
         )
     }
+
+    pub fn min(&self) -> Datum {
+        match self {
+            DataType::Int16 => Some(ScalarImpl::Int16(i16::MIN)),
+            DataType::Int32 => Some(ScalarImpl::Int32(i32::MIN)),
+            DataType::Int64 => Some(ScalarImpl::Int64(i64::MIN)),
+            DataType::Float32 => Some(ScalarImpl::Float32(OrderedF32::neg_infinity())),
+            DataType::Float64 => Some(ScalarImpl::Float64(OrderedF64::neg_infinity())),
+            DataType::Boolean => Some(ScalarImpl::Bool(false)),
+            DataType::Varchar => Some(ScalarImpl::Utf8("".to_string())),
+            DataType::Date => Some(ScalarImpl::NaiveDate(NaiveDateWrapper(NaiveDate::MIN))),
+            DataType::Time => Some(ScalarImpl::NaiveTime(NaiveTimeWrapper(
+                NaiveTime::from_hms(0, 0, 0),
+            ))),
+            DataType::Timestamp => Some(ScalarImpl::NaiveDateTime(NaiveDateTimeWrapper(
+                NaiveDateTime::MIN,
+            ))),
+            // FIXME(yuhao): Add a timestampz scalar.
+            DataType::Timestampz => Some(ScalarImpl::Int64(i64::MIN)),
+            DataType::Decimal => Some(ScalarImpl::Decimal(Decimal::NegativeInf)),
+            DataType::Interval => Some(ScalarImpl::Interval(IntervalUnit::min())),
+            DataType::Struct { .. } => Some(ScalarImpl::Struct(StructValue::new(vec![]))),
+            DataType::List { .. } => Some(ScalarImpl::List(ListValue::new(vec![]))),
+        }
+    }
```

Comment:
PTAL @xiangjinwu

Reply:
We may need to recursively call this function to match the schema of `Struct`.